github workflow test.yaml: test on ubuntu-latest failed

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->

https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835
```
sudo apt-get install libgpgme-dev libassuan-dev libbtrfs-dev libdevmapper-dev pkg-config -y
  git clone --depth 1 https://github.com/containers/skopeo -b v1.9.0 "$GITHUB_WORKSPACE"/src/github.com/containers/skopeo
  cd "$GITHUB_WORKSPACE"/src/github.com/containers/skopeo
  make bin/skopeo
  sudo install -D -m 755 bin/skopeo /usr/bin/skopeo
  rm -rf "$GITHUB_WORKSPACE"/src/github.com/containers/skopeo
  skopeo --version
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    LC_ALL: en_US.UTF-8
Reading package lists...
Building dependency tree...
Reading state information...
pkg-config is already the newest version (1.8.1-2build1).
The following additional packages will be installed:
  libbtrfs0t64 libgpg-error-dev libgpg-error-l10n libgpg-error0 libpcre2-16-0
  libpcre2-[3](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:3)2-0 libpcre2-8-0 libpcre2-dev libpcre2-posix3 libselinux1
  libselinux1-dev libsepol-dev libudev-dev
The following NEW packages will be installed:
  libassuan-dev libbtrfs-dev libbtrfs0t6[4](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:4) libdevmapper-dev libgpg-error-dev
  libgpgme-dev libpcre2-16-0 libpcre2-32-0 libpcre2-dev libpcre2-posix3
  libselinux1-dev libsepol-dev libudev-dev
The following packages will be upgraded:
  libgpg-error-l10n libgpg-error0 libpcre2-8-0 libselinux1
4 upgraded, 13 newly installed, 0 to remove and 74 not upgraded.
Need to get 2,772 kB of archives.
After this operation, 8,800 kB of additional disk space will be used.
Get:1 file:/etc/apt/apt-mirrors.txt Mirrorlist [142 B]
Get:2 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libgpg-error-l10n all 1.47-3build2.1 [8,146 B]
Get:3 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libgpg-error0 amd64 1.47-3build2.1 [70.1 kB]
Get:4 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libpcre2-8-0 amd64 10.42-4ubuntu2.1 [227 kB]
Get:[5](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:5) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd[6](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:6)4 libselinux1 amd64 3.5-2ubuntu2.1 [79.7 kB]
Get:6 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libgpg-error-dev amd64 1.4[7](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:7)-3build2.1 [128 kB]
Get:7 http://azure.archive.ubuntu.com/ubuntu noble/main amd64 libassuan-dev amd64 2.5.6-1build1 [95.0 kB]
Get:[8](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:8) http://azure.archive.ubuntu.com/ubuntu noble/main amd64 libbtrfs0t64 amd64 6.6.3-1.1build2 [13.3 kB]
Get:[9](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:9) http://azure.archive.ubuntu.com/ubuntu noble/main amd64 libbtrfs-dev amd64 6.6.3-1.1build2 [49.0 kB]
Ign:[10](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:10) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libudev-dev amd64 255.4-1ubuntu8.4
Get:[11](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:11) http://azure.archive.ubuntu.com/ubuntu noble/main amd64 libsepol-dev amd64 3.5-2build1 [384 kB]
Get:[12](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:13) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libpcre2-16-0 amd64 10.42-4ubuntu2.1 [210 kB]
Get:[13](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:14) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libpcre2-32-0 amd64 10.42-4ubuntu2.1 [197 kB]
Get:[14](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:15) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libpcre2-posix3 amd64 10.42-4ubuntu2.1 [6,748 B]
Get:[15](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:16) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libpcre2-dev amd64 10.42-4ubuntu2.1 [736 kB]
Get:[16](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:17) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libselinux1-dev amd64 3.5-2ubuntu2.1 [164 kB]
Get:[17](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:18) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 libdevmapper-dev amd64 2:1.02.[18](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:19)5-3ubuntu3.1 [36.7 kB]
Get:18 http://azure.archive.ubuntu.com/ubuntu noble/main amd64 libgpgme-dev amd64 1.18.0-4.1ubuntu4 [345 kB]
Ign:10 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 libudev-dev amd64 255.4-1ubuntu8.4
Err:10 http://security.ubuntu.com/ubuntu noble-updates/main amd64 libudev-dev amd64 255.4-1ubuntu8.4
  404  Not Found [IP: 52.147.2[19](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:20).192 80]
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/s/systemd/libudev-dev_255.4-1ubuntu8.4_amd64.deb  404  Not Found [IP: 52.147.[21](https://github.com/instructlab/instructlab/actions/runs/13067881217/job/36463386883?pr=2835#step:8:22)9.192 80]
Fetched 2,750 kB in 0s (6,372 kB/s)
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Error: Process completed with exit code 100.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**Device Info (please complete the following information):**
 - Hardware Specs: [e.g. Apple M2 Pro Chip, 16 GB Memory, etc.]
 - OS Version: [e.g. Mac OS 14.4.1, Fedora Linux 40]
 - Python Version: [output of `python --version`]
 - InstructLab Version: [output of `ilab system info`]

**Additional context**
<!-- Add any other context about the problem here. -->

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -84,6 +84,13 @@ jobs:
       - name: Free disk space
         uses: ./.github/actions/free-disk-space
 
+      # Always apt-get update before installing any extra packages
+      # https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/customizing-github-hosted-runners#installing-software-on-ubuntu-runners
+      - name: Update apt index
+        if: startsWith(matrix.platform, 'ubuntu')
+        run: |
+          sudo apt-get update
+
       - name: Install the expect package
         if: startsWith(matrix.platform, 'ubuntu')
         run: |

